File: main_project/OceCxxAdapter/src/FriendInputAdapter.cpp
#include "FriendInputAdapter.h"
#include "boost/lexical_cast.hpp"
#include <iostream>
//using namespace mop::hi::svc::adapter;
using namespace std;
using namespace com::xiaonei::services::FriendInput;
using namespace MyUtil;

//---------------------------------------------------------------------------
FriendInputManagerPrx FriendInputAdapter::getFriendInputManager(int id) {
	return locate<FriendInputManagerPrx>(_managers, "Manager", id, TWO_WAY);
}

FriendInputManagerPrx FriendInputAdapter::getFriendInputManagerOneway(int id) {
	return locate<FriendInputManagerPrx>(_managersOneway, "Manager", id, ONE_WAY);
}

void FriendInputAdapter::reLoadFriend(int id) {
	MCE_INFO("FriendInputAdapter -> reLoadFriend "<<id);
	getFriendInputManagerOneway(id)->reLoadFriend(id);
}

vector<int> FriendInputAdapter::getFriendInput(int hostId, string condition, int limit) {
  FriendInputManagerPrx proxy = getFriendInputManager(hostId);
  std::cout<< "enter into try!"<< std::endl;
  //try {
    std::cout<<"id:"<<hostId<<" condition:"<<condition<<std::endl;
    //proxy->getFriendInput(hostId, condition,limit);
    vector<int> temp = proxy->getFriendInput(hostId, condition,limit);
    std::cout << "hello, getFriendInput, proxy="<<proxy<< std::endl;
    return temp;//proxy->getFriendInput(hostId, condition,limit);
 // } catch (IceUtil::Exception& e) {
    //MCE_INFO("FriendInputAdapter::getFriendInput Exception:" << e.what() << "proxy=" << proxy);
 // }
 /* } catch (ConnectTimeoutException e) {
    Oce.getLogger().error(
        this.getClass().getName() + ".getFriendInput " + e);
  } catch (TimeoutException e) {
    Oce.getLogger().error(
        this.getClass().getName() + ".getFriendInput " + e);
  }*/
  //return _value;
}

File: main_project/client/java/OceAdapter/src/main/java/com/xiaonei/xce/friendinput/FriendInputAdapter.java
package com.xiaonei.xce.friendinput;

import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;
import java.util.Vector;

import xce.util.channel.Channel;
import xce.util.service.ServiceAdapter;

import Ice.ConnectTimeoutException;
import Ice.TimeoutException;

import com.xiaonei.services.FriendInput.*;
import com.xiaonei.xce.log.Oce;

public class FriendInputAdapter extends ServiceAdapter {
	private static FriendInputAdapter _instance = new FriendInputAdapter();
	private static final String endpoints = "@FriendInput";

	private Vector<FriendInputManagerPrx> _managers = new Vector<FriendInputManagerPrx>();
	private Vector<FriendInputManagerPrx> _managersOneway = new Vector<FriendInputManagerPrx>();

	protected FriendInputAdapter() {
		super(endpoints, 10, Channel.newGadgetsChannel(endpoints));

	}

	public static FriendInputAdapter getInstance() {
		return _instance;
	}

	protected FriendInputManagerPrx getFriendInputManager(int userId) {
		return FriendInputManagerPrxHelper.uncheckedCast(locateProxy(_managers,
				"Manager", userId, Channel.Invoke.Twoway,
				FriendInputManagerPrxHelper.class, 300));
	}

	protected FriendInputManagerPrx getFriendInputManagerOneway(int userId) {
		return FriendInputManagerPrxHelper.uncheckedCast(locateProxy(
				_managersOneway, "Manager", userId, Channel.Invoke.Oneway,
				FriendInputManagerPrxHelper.class, 300));
	}

	public int[] getFriendInput(int hostId, String condition, int limit) {
		int[] x = null;
		FriendInputManagerPrx prx = null;
		try {
			prx = getFriendInputManager(hostId);
			x = prx.getFriendInput(hostId, condition, limit);
		} catch (ConnectTimeoutException e) {
			Oce.getLogger().error(
					this.getClass().getName() + ".getFriendInput [" + prx
							+ "] userId=" + hostId + " " + e);
		} catch (TimeoutException e) {
			Oce.getLogger().error(
					this.getClass().getName() + ".getFriendInput [" + prx
							+ "] userId=" + hostId + " " + e);
		}
		return x;
	}

	public static void main(String[] args) {
		try {

			Scanner cin = new Scanner(System.in);
			String str = cin.nextLine();
			String[] s = str.split(" ");
			for (int i = 0; i < s.length; i++) {
				int[] x = FriendInputAdapter.getInstance().getFriendInput(
						220840375, s[i], 10);
				System.out.println(x.length);
				for (int u : x)
					System.out.println(u);
			}
		} catch (Exception e) {
			System.out.println(e);
		}

		System.exit(0);
	}
}
